.github/workflows/code_reviewer.yml

```diff
@@ -0,0 +1,21 @@
+name: AI PR Reviewer
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+permissions: write-all
```

Comment:
The permission `write-all` is quite broad. Consider specifying more granular permissions to adhere to the principle of least privilege. Refer to [GitHub's permissions documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) for more details.